### PR TITLE
[Bug] BracesFixer - Do add a line before close tag

### PR DESCRIPTION
**Pull request by SpacePossum**
Test for https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/1829

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
This will surely break most of our fixers?

**Comment by SpacePossum** [comment]
I ran 1.12 with the default configuration of the this project on the sample code and only the `BracesFixer` touched. Maybe we should add this case to the SF2 integration test, not sure though..

**Comment by GrahamCampbell** [comment]
When I said "this", I was more referring to that style of code, rather than that specific test case.

File: Symfony/CS/Tests/Fixer/PSR2/BracesFixerTest.php [hunk 1]
```diff
@@ -32,6 +32,20 @@ public function provideFixControlContinuationBracesCases()
...
             array(
                 '<?php
+class Foo
+{
+    public function A()
+    {
+        ?>
+        Test<?php echo $foobar; ?>Test
+        <?php
+        $a = 1;
```
Comment (SpacePossum) [inline Symfony/CS/Tests/Fixer/PSR2/BracesFixerTest.php]: When this line is removed, the fixer will add a white line.
The fix for this has a big impact on the performance of the fixer. If we've a transformer for the open tag this should not be an issue at all.

Comment (keradus) [inline Symfony/CS/Tests/Fixer/PSR2/BracesFixerTest.php]: the problem is that adding new tokens to collection is slow, and doing it in transformer means doing it for all code.

Comment (SpacePossum) [inline Symfony/CS/Tests/Fixer/PSR2/BracesFixerTest.php]: I understand, I propose to doc. this as a known edge case and not fix it, i.e. choose speed over fixing an edge case.

**Comment by SpacePossum** [comment]
rebased, RTM if test are green

**Comment by keradus** [comment]
btw, 1.12 not 1.11? why?

**Comment by SpacePossum** [comment]
my mistake, I tried and the commit can be picked as is on 1.11 without any problems.

**Comment by keradus** [comment]
OK, great !

**Comment by keradus** [comment]
Thank you @SpacePossum.

**Pull request closed without merging**